src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -97,47 +96,55 @@ Json::Value doTx (RPC::Context& context)
     if (!isHexTxID (txid))
         return rpcError (rpcNOT_IMPL);
 
-    auto te = context.app.getMasterTransaction ().fetch (
+    std::array<RangeSet<std::uint32_t>, 2> ledgers;
+
+    ledgers[0] = context.app.getLedgerMaster ().getCompleteLedgersRange ();
+
+    auto txn = context.app.getMasterTransaction ().fetch (
```

Comment:
Unfortunately, this doesn't handle the case from `LedgerMaster::setFullLedger`. That adds the ledger to the history range and then queue the update to the sqlite database to the job queue. That can violate our invariant that the range computed here is a strict subset of the searched ledgers.

I thought we could address that by computing the min and max ledgers in the database, and intersecting with that as well. But that only works if `setFullLedger` is only ever called with ledgers the increase the sequence number. It looks like that assumption isn't true. I need to think about how to address this.

Reply:
Ok